### PR TITLE
[Frontend]Implement to `publicRuntimeConfig` for that BASE_URL and KUBE_API_SERVER_URL can be set in env for frontend image 

### LINKED ISSUE #150: Fix to `BASE_URL` and `KUBE_API_SERVER_URL` can be set in env for frontend image
## Problem
Currently, Nuxtjs(frontend framework) can be set `BASE_URL` and `KUBE_API_SERVER_URL` via `.env` file.
However, it is better to specify both values from the env in the manifest file (such as `docker-compose.yaml`).

/priority important-soon
/kind feature
/assign

### PR DESCRIPTION
**Pull request by 196Ikuchil**
#### What type of PR is this?
/kind bug
/kind feature

#### What this PR does / why we need it:
Currently, Nuxtjs(frontend framework) can be set BASE_URL and KUBE_API_SERVER_URL via .env file.
We fix it so that we can set these `env` in runtime.

#### Which issue(s) this PR fixes:

Fixes #150

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: web/Dockerfile [hunk 1]
```diff
@@ -19,11 +19,16 @@ WORKDIR /app
 
 ENV NODE_ENV production
 ENV NUXT_TELEMETRY_DISABLED 1
+# Currnently, HOST_ENV is used to decide whether to import the TEMPLATE files of the k8s resources.
+# ./components/lib/templates/*.yaml
+ENV HOST_ENV production
```
Comment (196Ikuchil) [inline web/Dockerfile:24]: See also `web/nuxt.config.js`. 
Is there a better way? (These TEMPLATEs files are only needed at build time. Also, these env values must not be present at the application startups. )

File: web/Dockerfile [hunk 2]
```diff
@@ -19,11 +19,16 @@ WORKDIR /app
...
+ENV BASE_URL http://localhost:1212
+ENV KUBE_API_SERVER_URL http://localhost:3131
 
 RUN addgroup -g 1001 -S nodejs
 RUN adduser -S nuxtjs -u 1001
 
-#COPY --from=builder /app/nuxt.config.js ./
+COPY --from=builder /app/nuxt.config.js ./
```
Comment (196Ikuchil) [inline web/Dockerfile:31]: Required

**Comment by 196Ikuchil** [comment]
Re-implemented logic for reading runtimeconfig.

File: web/api/APIProvider.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { PriorityClassAPIKey } from "./APIProviderKeys";
+import { ExportAPIKey } from "./APIProviderKeys";
+import { PVAPIKey } from "./APIProviderKeys";
+import { PVCAPIKey } from "./APIProviderKeys";
+import { ResetAPIKey } from "./APIProviderKeys";
+import { SchedulerconfigurationAPIKey } from "./APIProviderKeys";
+import { StorageClassAPIKey } from "./APIProviderKeys";
+
+export default defineComponent({
+  setup() {
```
Comment (196Ikuchil) [inline web/api/APIProvider.vue:29]: > setup() itself does not have access to the component instance 
https://vuejs.org/api/composition-api-setup.html#basic-usage

Therefore, I think that we can't access `$config` and we must use `Plugin` to access its value. 
 (Note: In `Nuxt3`, we maybe use `useRuntimeConfig()` function.)

File: web/plugins/ApiRuntimeConfigPlugin.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Plugin } from "@nuxt/types";
+
+const ApiRuntimeConfigPlugin: Plugin = (context, inject): void => {
```
Comment (196Ikuchil) [inline web/plugins/ApiRuntimeConfigPlugin.ts:3]: A `Plugin` will be loaded at an earlier time in the Nuxt lifecycle.

**Comment by 196Ikuchil** [comment]
Implementation is almost complete.
However, #164 changes need to be incorporated.

/hold

**Comment by 196Ikuchil** [comment]
@sanposhiho Thanks! Fixed it.

**Review by sanposhiho — APPROVED** [review]
Congrats 🎉 

/lgtm
/approve

**Pull request merged**